src/walletConnect/analytics.ts

```diff
@@ -0,0 +1,96 @@
+import '@react-native-firebase/database'
+import '@react-native-firebase/messaging'
+import { SessionTypes, SignClientTypes } from '@walletconnect/types'
+import { getDappRequestOrigin } from 'src/app/utils'
+import { ActiveDapp } from 'src/dapps/types'
+import {
+  WalletConnectPayloadRequest,
+  WalletConnectSession,
+  WalletConnectSessionRequest,
+} from 'src/walletConnect/types'
+
+const isSessionProposalType = (
+  session: SignClientTypes.EventArguments['session_proposal'] | SessionTypes.Struct
+): session is SignClientTypes.EventArguments['session_proposal'] => {
+  return 'params' in session
+}
+
+export function getDefaultSessionTrackedPropertiesV2(
+  session: SignClientTypes.EventArguments['session_proposal'] | SessionTypes.Struct,
+  activeDapp: ActiveDapp | null
+) {
+  const peer = isSessionProposalType(session) ? session.params.proposer : session.peer
+  const { name: dappName, url: dappUrl, description: dappDescription, icons } = peer.metadata
+
+  const relayProtocol = isSessionProposalType(session)
+    ? session.params.relays[0]?.protocol
+    : session.relay.protocol
+
+  const requiredNamespaces = isSessionProposalType(session)
+    ? session.params.requiredNamespaces
+    : session.requiredNamespaces
+
+  return {
+    version: 2 as const,
+    dappRequestOrigin: getDappRequestOrigin(activeDapp),
+    dappName,
+    dappUrl,
+    dappDescription,
+    dappIcon: icons[0],
+    relayProtocol,
+    ...Object.keys(requiredNamespaces).reduce((acc, key) => {
+      const { chains, events, methods } = requiredNamespaces[key]
+      return {
+        ...acc,
+        [`${key}Events`]: events,
+        [`${key}Chains`]: chains,
+        [`${key}Methods`]: methods,
+      }
+    }, {}),
+  }
+}
+
+export function getDefaultRequestTrackedPropertiesV2(
+  request: SignClientTypes.EventArguments['session_request']
+) {
+  const { id, params } = request
+
+  return {
+    requestChainId: params.chainId,
+    requestId: id,
+    requestJsonrpc: '2.0',
```

Comment:
this was previously params.request.params which is wrong, i don't think this field is so relevant in v2 anymore